### ISSUE TITLE
Checkpointing and serialization (orbax integration)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/somax#27]
## Summary

Long simulations need checkpoint-to-disk for restarts. Define the checkpointing strategy using orbax-checkpoint and document the gradient checkpointing options from diffrax.

Parent: Sub-epic: Core API Framework

### Disk checkpointing (orbax)

```python
import orbax.checkpoint as ocp

checkpointer = ocp.StandardCheckpointer()
checkpointer.save(path / "step_1000", {"state": state, "params": model.params})
ckpt = checkpointer.restore(path / "step_1000", target=...)
```

**orbax-checkpoint** is the primary choice — JAX-native pytree serialization maintained by Google/DeepMind.

**eqx.tree_serialise_leaves** is a simpler alternative for small models.

**DVC** versions checkpoint files for reproducibility.

### Gradient checkpointing (AD through long windows)

diffrax provides several adjoint methods:

| Adjoint | Memory | Gradients | Use case |
|---------|--------|-----------|----------|
| `DirectAdjoint` | O(N) | Exact | Default, short windows |
| `RecursiveCheckpointAdjoint` | O(sqrt(N)) | Exact | Long training windows |
| `BacksolveAdjoint` | O(1) | Approximate | Very long windows, memory-critical |

## Acceptance Criteria

- [ ] Save/restore model state and params via orbax
- [ ] Integration with DVC pipeline outputs
- [ ] Document which diffrax adjoint to use for each scenario
- [ ] Example: restart a simulation from checkpoint
